test/integration/ads_integration_test.cc

```diff
@@ -261,34 +261,40 @@ TEST_P(AdsIntegrationTest, DeltaSdsRemovals) {
             name: validation_context
             sds_config:
               resource_api_version: V3
+              initial_fetch_timeout: 5s
               ads: {}
   )EOF",
                             sds_transport_socket);
   auto cluster = ConfigHelper::buildStaticCluster("cluster", 8000, "127.0.0.1");
   *cluster.mutable_transport_socket() = sds_transport_socket;
   cluster.set_name("cluster_0");
 
-  // Initial * Delta ADS subscription
+  // Initial * Delta ADS subscription.
   EXPECT_TRUE(compareDeltaDiscoveryRequest(cds_type_url, {}, {}));
   sendDeltaDiscoveryResponse<envoy::config::cluster::v3::Cluster>(cds_type_url, {cluster}, {}, "1");
 
-  // The cluster needs this secret, so it's going to request it
+  // The cluster needs this secret, so it's going to request it.
   EXPECT_TRUE(compareDeltaDiscoveryRequest(sds_type_url, {"validation_context"}, {}, {}));
 
+  // Cluster should start off warming as the secret is being requested.
+  test_server_->waitForGaugeEq("cluster.cluster_0.warming_state", 1);
+
   // Ack the original CDS sub
   EXPECT_TRUE(compareDeltaDiscoveryRequest(cds_type_url, {}, {}));
 
-  // Before we send the secret, we'll send a delta removal to make sure we don't get a NACK
+  // Before we send the secret, we'll send a delta removal to make sure we don't get a NACK.
   sendDeltaDiscoveryResponse<envoy::extensions::transport_sockets::tls::v3::Secret>(
-      sds_type_url, {}, {"validation_context"}, "2");
+      sds_type_url, {}, {"validation_context"}, "1");
 
-  // Ack the removal
-  EXPECT_TRUE(compareDeltaDiscoveryRequest(sds_type_url, {}, {}));
+  // The cluster shouldn't be warming anymore since the server signaled
+  // that the requested resource doesn't exist.
+  test_server_->waitForGaugeEq("cluster.cluster_0.warming_state", 0);
 
-  // Cluster should still be warming
-  test_server_->waitForGaugeGe("cluster_manager.warming_clusters", 1);
-  test_server_->waitForGaugeEq("cluster.cluster_0.warming_state", 1);
-  test_server_->waitForCounterEq("cluster_manager.cluster_removed", 0);
+  // Warming should finish, so expect LDS ACK.
```

Comment:
RE `Warming should finish, so expect LDS ACK.`, I don't think there should be a correlation between warming state and sending an ACK/NACK, as those are 2 different things (it may be that it wasn't observed in the prior test because this is the initialization/startup phase of Envoy).
Consider rephrasing to something like "Ack the original LDS subscription."